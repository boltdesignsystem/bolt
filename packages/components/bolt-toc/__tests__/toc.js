import { render, stopServer } from '../../../testing/testing-helpers';

describe('<bolt-toc> Component', () => {
  afterAll(async () => {
    await stopServer();
  }, 100);

  // Basic Usage
  test('Basic usage', async () => {
    const results = await render('@bolt-components-toc/toc.twig', {
      items: [
        {
          text: 'Section One',
          url: '#section-one',
        },
        {
          text: 'Section Two',
          url: '#section-two',
        },
        {
          text: 'Section Three',
          url: '#section-three',
        },
      ],
    });
    expect(results.ok).toBe(true);
    expect(results.html).toMatchSnapshot();
  });

  // Props
  test(`Table of content with a header`, async () => {
    const results = await render('@bolt-components-toc/toc.twig', {
      header: 'This is the header',
      items: [
        {
          text: 'Section One',
          url: '#section-one',
        },
        {
          text: 'Section Two',
          url: '#section-two',
        },
        {
          text: 'Section Three',
          url: '#section-three',
        },
      ],
    });
    expect(results.ok).toBe(true);
    expect(results.html).toMatchSnapshot();
  });

  test(`Manuualy set the current item`, async () => {
    const results = await render('@bolt-components-toc/toc.twig', {
      items: [
        {
          text: 'Section One',
          url: '#section-one',
        },
        {
          text: 'Section Two',
          url: '#section-two',
          active: true,
        },
        {
          text: 'Section Three',
          url: '#section-three',
        },
      ],
    });
    expect(results.ok).toBe(true);
    expect(results.html).toMatchSnapshot();
  });
});
