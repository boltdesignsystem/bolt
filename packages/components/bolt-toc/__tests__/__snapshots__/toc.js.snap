// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<bolt-toc> Component Basic usage 1`] = `
<bolt-toc>
  <div class="c-bolt-toc c-bolt-toc--variant-card"
       role="navigation"
       aria-labelledby="js-bolt-toc-12345"
  >
    <strong class="c-bolt-toc__header c-bolt-toc__header--hidden"
            id="js-bolt-toc-12345"
    >
      Table of Content
    </strong>
    <div class="c-bolt-toc__list"
         role="list"
    >
      <ssr-keep for="bolt-toc"
                role="presentation"
      >
        <bolt-toc-item url="#section-one"
                       role="presentation"
        >
          <div role="listitem">
            <a class="c-bolt-toc-item "
               href="#section-one"
            >
              <ssr-keep for="bolt-toc-item">
                Section One
              </ssr-keep>
            </a>
          </div>
        </bolt-toc-item>
        <bolt-toc-item url="#section-two"
                       role="presentation"
        >
          <div role="listitem">
            <a class="c-bolt-toc-item "
               href="#section-two"
            >
              <ssr-keep for="bolt-toc-item">
                Section Two
              </ssr-keep>
            </a>
          </div>
        </bolt-toc-item>
        <bolt-toc-item url="#section-three"
                       role="presentation"
        >
          <div role="listitem">
            <a class="c-bolt-toc-item "
               href="#section-three"
            >
              <ssr-keep for="bolt-toc-item">
                Section Three
              </ssr-keep>
            </a>
          </div>
        </bolt-toc-item>
      </ssr-keep>
    </div>
  </div>
</bolt-toc>
`;

exports[`<bolt-toc> Component Manuualy set the current item 1`] = `
<bolt-toc>
  <div class="c-bolt-toc c-bolt-toc--variant-card"
       role="navigation"
       aria-labelledby="js-bolt-toc-12345"
  >
    <strong class="c-bolt-toc__header c-bolt-toc__header--hidden"
            id="js-bolt-toc-12345"
    >
      Table of Content
    </strong>
    <div class="c-bolt-toc__list"
         role="list"
    >
      <ssr-keep for="bolt-toc"
                role="presentation"
      >
        <bolt-toc-item url="#section-one"
                       role="presentation"
        >
          <div role="listitem">
            <a class="c-bolt-toc-item "
               href="#section-one"
            >
              <ssr-keep for="bolt-toc-item">
                Section One
              </ssr-keep>
            </a>
          </div>
        </bolt-toc-item>
        <bolt-toc-item url="#section-two"
                       active
                       role="presentation"
        >
          <div role="listitem">
            <a class="c-bolt-toc-item c-bolt-toc-item--current"
               href="#section-two"
            >
              <ssr-keep for="bolt-toc-item">
                Section Two
              </ssr-keep>
            </a>
          </div>
        </bolt-toc-item>
        <bolt-toc-item url="#section-three"
                       role="presentation"
        >
          <div role="listitem">
            <a class="c-bolt-toc-item "
               href="#section-three"
            >
              <ssr-keep for="bolt-toc-item">
                Section Three
              </ssr-keep>
            </a>
          </div>
        </bolt-toc-item>
      </ssr-keep>
    </div>
  </div>
</bolt-toc>
`;

exports[`<bolt-toc> Component Table of content with a header 1`] = `
<bolt-toc header="This is the header">
  <div class="c-bolt-toc c-bolt-toc--variant-card"
       role="navigation"
       aria-labelledby="js-bolt-toc-12345"
  >
    <strong class="c-bolt-toc__header"
            id="js-bolt-toc-12345"
    >
      This is the header
    </strong>
    <div class="c-bolt-toc__list"
         role="list"
    >
      <ssr-keep for="bolt-toc"
                role="presentation"
      >
        <bolt-toc-item url="#section-one"
                       role="presentation"
        >
          <div role="listitem">
            <a class="c-bolt-toc-item "
               href="#section-one"
            >
              <ssr-keep for="bolt-toc-item">
                Section One
              </ssr-keep>
            </a>
          </div>
        </bolt-toc-item>
        <bolt-toc-item url="#section-two"
                       role="presentation"
        >
          <div role="listitem">
            <a class="c-bolt-toc-item "
               href="#section-two"
            >
              <ssr-keep for="bolt-toc-item">
                Section Two
              </ssr-keep>
            </a>
          </div>
        </bolt-toc-item>
        <bolt-toc-item url="#section-three"
                       role="presentation"
        >
          <div role="listitem">
            <a class="c-bolt-toc-item "
               href="#section-three"
            >
              <ssr-keep for="bolt-toc-item">
                Section Three
              </ssr-keep>
            </a>
          </div>
        </bolt-toc-item>
      </ssr-keep>
    </div>
  </div>
</bolt-toc>
`;
